docs/api/DesktopAgent.md

```diff
@@ -215,6 +216,44 @@ var intentR = await agent.raiseIntent("intentName", context);
 var dataR = intentR.data;
 ```
 
+### `IntentListenerResponse`
+
+```typescript
+type IntentListenerResponse = Promise<{data:any}> | void;
+```
+
+IntentListenerResponse describes the return type of an intent listener handler
+set up with addIntentListener.
+
+It's valid to not return anything at all, for "fire and forget" type intents,
+otherwise the handler must return a Promise that resolves to an object with a
+"data" property in it.
+ 
+#### Example
+```javascript
+// Valid
+agent.addIntentListener("IntentName", (context) => {
+  doSomething();
+});
+agent.addIntentListener("IntentName2", (context) => {
+  return Promise.resolve({
+    data: payload
+  });
+});
+
+// Invalid
+agent.addIntentListener("IntentName", (context) => {
```

Comment:
I'm not sure there is any value in providing examples that invalidate the interface itself. Let TypeScript do its job!

If we include invalid examples, perhaps consider adding those which highlight issues that are not 'trapped' by the interface itself.